src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,225 @@
+        else:
+            logging.info("HoldTime not supported. The rest of legacy attribute test can be skipped")
+
+        self.step(7)
+        if attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list:
+            if ((occupancy_sensor_type_bitmap_dut == 0b00000001) | ((occupancy_sensor_type_bitmap_dut != 0b00000001)&(occupancy_sensor_type_bitmap_dut != 0b00000010)&(occupancy_sensor_type_bitmap_dut != 0b00000100))):
+                pir_otou_delay_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIROccupiedToUnoccupiedDelay)
+
+                asserts.assert_less_equal(pir_otou_delay_dut, 0xFFFE, "PIROccupiedToUnoccupiedDelay is not in valid range")
+                asserts.assert_greater_equal(pir_otou_delay_dut, 0, "PIROccupiedToUnoccupiedDelay is not in valid range")
+            else:
+                logging.info("PIROccupiedToUnoccupiedDelay conformance failed. Test step skipped")
+        else:
+            logging.info("PIROccupiedToUnoccupiedDelay not supported. Test step skipped")
+
+        self.step(8)
+        if attributes.PIRUnoccupiedToOccupiedDelay.attribute_id in attribute_list: 
+            if attributes.PIRUnoccupiedToOccupiedThreshold.attribute_id in attribute_list:
+            
+                pir_utoo_delay_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIRUnoccupiedToOccupiedDelay)
+                asserts.assert_less_equal(pir_utoo_delay_dut, 0xFFFE, "PIRUnoccupiedToOccupiedDelay is not in valid range")
+                asserts.assert_greater_equal(pir_utoo_delay_dut, 0, "PIRUnoccupiedToOccupiedDelay is not in valid range")
+            else:
+                logging.info("PIRUnoccupiedToOccupiedDelay conformance failed. Test step skipped")
+        else:
+            logging.info("PIRUnoccupiedToOccupiedDelay not supported. Test step skipped")
+
+        self.step(9)
+        if attributes.PIRUnoccupiedToOccupiedThreshold.attribute_id in attribute_list:
+            if attributes.PIRUnoccupiedToOccupiedDelay.attribute_id in attribute_list:                
+                pir_utoo_threshold_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIRUnoccupiedToOccupiedThreshold)
+    
+                asserts.assert_less_equal(pir_utoo_threshold_dut, 0xFE, "PIRUnoccupiedToOccupiedThreshold is not in valid range")
+                asserts.assert_greater_equal(pir_utoo_threshold_dut, 0, "PIRUnoccupiedToOccupiedThreshold is not in valid range")
+            else:
+                logging.info("PIRUnoccupiedToOccupiedThreshold conformance failed. Test step skipped")
+        else:
+            logging.info("PIRUnoccupiedToOccupiedThreshold not supported. Test step skipped")
+
+        self.step(10)
+        if attributes.UltrasonicOccupiedToUnoccupiedDelay.attribute_id in attribute_list:

```

Comment:
To do the forward and reverse conformance checks in one, suggest doing as above:
has_ultrasonic_bitmap = (occupancy_sensor_type_bitmap_dut & Clusters.OccupancySensing.Enums.OccupancySensorTypeEnum.kUltrasonic) != 0
has_ultrasonic_delay = attributes.UltrasonicOccupiedToUnoccupiedDelay.attribute_id in attribute_list

asserts.assert_equal(has_ultrasonic_bitmap, has_ultrasonic_delay, "Bad conformance on Ultrasonic bitmap")

Reply:
Done